File: solutions/lua/collatz-conjecture/1/collatz-conjecture.lua
return function(n)
    if n<=0 
    then
        error('Only positive numbers are allowed')
    end
    c=0
    while (n~=1)
    do 
        c=c+1
        if n % 2 == 0
        then
            n=n/2
        else
            n=3*n+1
        end
    end
    return c
end

File: solutions/javascript/collatz-conjecture/1/collatz-conjecture.js
//
// This is only a SKELETON file for the 'Collatz Conjecture' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const steps = (num) => {
  if (num <= 0) {
    throw new Error('Only positive numbers are allowed');
  }
  var steps = 0;
  while (num != 1) {
    steps++;
    if (num % 2 == 0) {
      num /= 2;
    } else {
      num = 3 * num + 1;
    }
  }
  return steps;
};
